pkg/modifiers/modifiers.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2024 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package modifiers provides the functionality to parse meta modifiers from text.
+package modifiers
+
+import (
+	"context"
+	"regexp"
+)
+
+var metaValuesKVRegex = regexp.MustCompile(`(?m)^(GUARDIAN_[A-Z0-9_]+)=(.*)$`)
```

Comment:
I'm not sure that multiline mode makes sense here, can you help me understand why? Referring to `(?m)`

It might be cleaner to iterate over the lines of `contents` and match each line individually. That avoids the risk of having a single "match" that unintentionally spans multiple lines.

Reply:
Mmmmm this is fair, updated.